### PR TITLE
Make combine_obsspace work for IODAv3 + Other Minor Changes

### LINKED ISSUE #1152: Combine Obsspace code needs updated to new naming conventions
## Description
As the title says, when switching to the new IODA conventions, the combine_obsspace code was not updated, so it is unable to combine conventional obs together.

## Requirements

Combine conventional obs files together (e.g. sondes_t, sondes_uv to sondes)

## Acceptance Criteria (Definition of Done)
When it is successfully able to combine obs spaces

## Dependencies
None

### PR DESCRIPTION
**Pull request by CoryMartin-NOAA**
## Description

This PR mainly re-enables the ability to combine IODA (and GeoVaLs) files produced through GSI for UFO science validation purposes. However, there are some minor additional variables added to support additional ob types (GNSSRO, SatWinds, Ozone, etc.).

While I'm issuing this PR, much of this work was performed by @emilyhcliu and supported through the efforts of others (e.g. @XuanliLi-NOAA and @BrettHoover-NOAA among others) and tested by @danholdaway 

### Issue(s) addressed

Link the issues to be closed with this PR
- fixes #1152 

## Acceptance Criteria (Definition of Done)

Approved by someone at both JCSDA and GMAO

## Dependencies

None

## Impact

None

### CODE REVIEW

**Review by BenjaminRuston — APPROVED** [review]
looks reasonable here, thanks Cory

**Comment by BenjaminRuston** [comment]
there are some coding norms to fix:

```


The following tests FAILED:
--
950 | 2 - iodaconv_gsi_ncdiag_coding_norms (Failed)
951 | 56 - test_iodaconv_gsidiag_conv_uv (Failed)
952 | 57 - test_iodaconv_gsidiag_conv (Failed)
953 | 58 - test_iodaconv_gsidiag_rad (Failed)

```

**Comment by CoryMartin-NOAA** [comment]
whoops, looks like I need to update test references too, will try to tackle that this afternoon, otherwise early next week.

**Review by danholdaway — APPROVED** [review]
Thanks so much for making these changes @CoryMartin-NOAA and @emilyhcliu. It's working for us as far as I can see.

**Review by gmao-jjin3 — APPROVED** [review]

**Comment by BenjaminRuston** [comment]
great, thanks for the updates @CoryMartin-NOAA

File: src/lib_python/ioda_conv_engines.py [hunk 1]
```diff
@@ -96,7 +96,7 @@ def WriteGeoVars(self, GeoVars, GeoVarDims, GeoVarAttrs):
                 dims = GeoVarDims[VarName]
             else:
                 # assume it is just nlocs
-                dims = ['Location']
+                dims = ['nlocs']
```
Comment (emilyhcliu) [inline src/lib_python/ioda_conv_engines.py:99]: My fault!! I missed this one.

Comment (BenjaminRuston) [reply src/lib_python/ioda_conv_engines.py:99]: ok,, this is odd to me is this for backward support?

Comment (CoryMartin-NOAA) [reply src/lib_python/ioda_conv_engines.py:99]: the geoval reader for the UFO ctests still assumes 'nlocs' (I believe) and not Location

Comment (BenjaminRuston) [reply src/lib_python/ioda_conv_engines.py:99]: ah yes,, thanks

**Review by srherbener — APPROVED** [review]
Thanks for make these updates!

**Pull request merged**